### ISSUE TITLE
attention_mask does not work when passed as kwarg

### ISSUE THREAD

**Issue by keltin13** [ndif-team/nnsight#542]
When you pass an attention mask to `model.generate` as a keyword argument, it appears that is getting ignored (or is otherwise not working properly). Is this expected behavior?

In the script below you can see that the model produces different responses for the first prompt (the shorter one that needs padding) when generate is called like `model.generate(inputs['input_ids'], attention_mask=inputs['attention_mask'])`, whereas passing the prompts directly as strings or as a single tokenized dictionary produce the same (correct) generations.

```py
from nnsight import LanguageModel

model = LanguageModel("google/gemma-2-2b-it", torch_dtype="float16", device_map="cuda:0", dispatch=True)
tokenizer = model.tokenizer

prompts = [
    'What are the best strategies for learning a new language?',
    'Write 3 questions that a student can ask their teacher about a given topic.'
]
messages = [
    tokenizer.apply_chat_template(
        [{"role": "user", "content": p}],
        tokenize=False, add_generation_prompt=True, bos_token=False
    )[len("<bos>"):]  # Prevent double <bos>
    for p in prompts
]
inputs = tokenizer(messages, padding=True, truncation=False, return_tensors="pt", add_special_tokens=True)

# Passing strings directly
with model.generate(messages, max_new_tokens=20, do_sample=False):
    out = model.generator.output.save()

for o in model.tokenizer.batch_decode(out):
    print(o)

print('\n\n' + '-' * 100 + '\n')

# Passing input_ids/attention_mask together
with model.generate(inputs, max_new_tokens=20, do_sample=False) as tracer:
    out = model.generator.output.save()

for o in model.tokenizer.batch_decode(out):
    print(o)

print('\n\n' + '-' * 100 + '\n')

# Passing input_ids/attention_mask separately
with model.generate(inputs['input_ids'], attention_mask=inputs['attention_mask'], max_new_tokens=20, do_sample=False) as tracer:
    out = model.generator.output.save()

for o in model.tokenizer.batch_decode(out):
    print(o)
```

```
<pad><pad><pad><pad><pad><bos><start_of_turn>user
What are the best strategies for learning a new language?<end_of_turn>
<start_of_turn>model
There's no single "best" strategy, as the most effective approach depends on your individual learning
<bos><start_of_turn>user
Write 3 questions that a student can ask their teacher about a given topic.<end_of_turn>
<start_of_turn>model
Please provide the **topic** so I can write 3 relevant questions for a student to ask their

----------------------------------------------------------------------------------------------------

<pad><pad><pad><pad><pad><bos><start_of_turn>user
What are the best strategies for learning a new language?<end_of_turn>
<start_of_turn>model
There's no single "best" strategy, as the most effective approach depends on your individual learning
<bos><start_of_turn>user
Write 3 questions that a student can ask their teacher about a given topic.<end_of_turn>
<start_of_turn>model
Please provide the **topic** so I can write 3 relevant questions for a student to ask their

----------------------------------------------------------------------------------------------------

<pad><pad><pad><pad><pad><bos><start_of_turn>user
What are the best strategies for learning a new language?<end_of_turn>
<start_of_turn>model
Here's a breakdown of effective strategies for learning a new language, combining the best practices from various
<bos><start_of_turn>user
Write 3 questions that a student can ask their teacher about a given topic.<end_of_turn>
<start_of_turn>model
Please provide the **topic** so I can write 3 relevant questions for a student to ask their
```
